teos/src/config.rs

```diff
@@ -242,6 +242,30 @@ impl Config {
     pub fn is_default(&self) -> bool {
         self == &Config::default()
     }
+
+    /// Logs non-default options.
+    pub fn log_non_default_options(&self) {
+        let json_default_config = serde_json::json!(&Config::default());
+        let json_config = serde_json::json!(&self);
+        let protected_keys = ["btc_rpc_user", "btc_rpc_password"];
+
+        for (key, value) in json_config.as_object().unwrap().iter() {
+            if *value != json_default_config[key] {
+                log::info!(
+                    "Custom config arg: {}: {}",
+                    key,
+                    if protected_keys.contains(&key.as_str()) {
+                        "****".to_owned()
+                    } else {
+                        value.to_string()
+                    }
+                );
+            }
+            if *value != json_default_config[key] {
```

Comment:
You forgot to remove this. Things are being logged twice now.

Do not rush pushing things, please do test things first.